Test fails without retrying on concurrent modifications.

**Describe the bug**
Related to #3599 we found this odd failure that looks like updateOrFail that fails the test instead of retrying for concurrent mods.

```
    broker_control_plane_test_helper.go:135: Error: Could not update trigger trigger-no-broker: Operation cannot be fulfilled on triggers.eventing.knative.dev "trigger-no-broker": the object has been modified; please apply your changes to the latest version and try again
        --- FAIL: TestBrokerV1Beta1ControlPlane/Channel-messaging.knative.dev/v1beta1/Ready_Trigger_V1Beta1_(no_Broker)_set_Broker_and_includes_status.subscriber_Uri (0.16s)
```

 **Expected behavior**
Tests to always pass :) 

 **To Reproduce**
Not sure, just saw in the test logs.

 **Knative release version**
HEAD

 **Additional context**
Add any other context about the problem here such as proposed priority

--- a/test/conformance/helpers/broker_control_plane_test_helper.go
+++ b/test/conformance/helpers/broker_control_plane_test_helper.go
@@ -23,6 +23,7 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 	eventingv1beta1 "knative.dev/eventing/pkg/apis/eventing/v1beta1"
+	"knative.dev/pkg/reconciler"
 
 	testlib "knative.dev/eventing/test/lib"
 	"knative.dev/eventing/test/lib/duck"
@@ -130,7 +131,11 @@ func TriggerV1Beta1ReadyAfterBrokerIncludesSubURI(t *testing.T, triggerName, bro
 		t.Fatalf("Error: Could not get trigger %s: %v", triggerName, err)
 	}
 	tr.Spec.Broker = brokerName
-	_, err = client.Eventing.EventingV1beta1().Triggers(client.Namespace).Update(tr)
+	err = reconciler.RetryUpdateConflicts(func(attempts int) (err error) {
+		_, e := client.Eventing.EventingV1beta1().Triggers(client.Namespace).Update(tr)
+		return e
+	})
+
 	if err != nil {
 		t.Fatalf("Error: Could not update trigger %s: %v", triggerName, err)
 	}

